cwltool/pathmapper.py

```diff
@@ -139,6 +144,30 @@ def trim_listing(obj):
     if obj.get("location", "").startswith("file://") and "listing" in obj:
         del obj["listing"]
 
+# Download http Files
+def downloadHttpFile(httpurl):
+    cache_session = None
+    if "HOME" in os.environ:
```

Comment:
I made these changes, i assumed you meant `$HOME/.cache` when `XDG_CACHE_HOME` is not present